test/distributed/test_c10d_gloo.py

```diff
@@ -443,6 +443,18 @@ def test_allreduce_checks(self):
             opts = c10d.AllreduceOptions()
             pg.allreduce([t1, t3], opts)
 
+    @requires_gloo()
+    def test_allreduce_timeout(self):
```

Comment:
maybe add a test case where the collective timeout takes precedence over the pg timeout thats set?